File: src/main/java/victor/training/cleancode/SplitPhase.java
package victor.training.cleancode;

import org.jetbrains.annotations.NotNull;

import java.util.Collections;
import java.util.Map;

public class SplitPhase {
    public float calculateOrderPrice(String orderString, Map<String, Integer> priceList) {
        OrderLine orderLine = parse(orderString);
        return computePrice(orderLine, priceList);
    }

    private static int computePrice(OrderLine result, Map<String, Integer> priceList) {
        Integer productPrice = priceList.get(result.productCode());
        return result.quantity() * productPrice;
    }

    @NotNull
    private static OrderLine parse(String orderString) {
        String[] orderData = orderString.split("\\s+");
        String productCode = orderData[0].split("-")[1];
        int quantity = Integer.parseInt(orderData[1]);
        OrderLine result = new OrderLine(productCode, quantity);
        return result;
    }

    private record OrderLine(String productCode, int quantity) {
    }

    public static void main(String[] args) {
        System.out.println(new SplitPhase().calculateOrderPrice("Chair-CHR 4",
                Collections.singletonMap("CHR", 5)));
    }
}

File: src/main/java/victor/training/cleancode/SplitPhase.kt
package victor.training.cleancode

data class ParsedOrder(val productCode:String, val items:Int)
fun calculateOrderPrice(orderString: String, priceList: Map<String, Int>): Float {
    val parsedOrder = parse(orderString)
    return computePrice(priceList, parsedOrder)
}

private fun computePrice(
    priceList: Map<String, Int>,
    parsedOrder: ParsedOrder
): Float {
    val productPrice = priceList[parsedOrder.productCode]!!
    return (parsedOrder.items * productPrice).toFloat()
}

private fun parse(orderString: String): ParsedOrder {
    val orderData = orderString.split("\\s+".toRegex())
    val productCode = orderData[0].split("-".toRegex())[1]
    val productItems = orderData[1].toInt()

    return ParsedOrder(productCode, productItems)
}

fun main() {
    println(calculateOrderPrice("Chair-CHR 4", mapOf("CHR" to 5)))
}